src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -90,6 +95,19 @@ private ITypeSymbol GetSymbolType(ISymbol symbol)
                 supportedPlatforms: supportedPlatformData);
         }
 
+        private static bool ShouldSoftSelectInArgumentList(CompletionContext completionContext, SyntaxContext context, bool preselect)
+        {
+            return !preselect &&
+                completionContext.Trigger.Kind == CompletionTriggerKind.Insertion &&
+                context.IsOnArgumentListBracketOrComma &&
+                IsArgumentListTriggerCharacter(completionContext.Trigger.Character);
+        }
+
+        private static bool IsArgumentListTriggerCharacter(char character)
+        {
+            return character == ' ' || character == '(' || character == '[';
```

Comment:
dupe?  use helper?

Reply:
There's no shared CompletionUtilities, so I wasn't sure where to put it.

Reply:
:'(